source/server/http/admin.cc

```diff
@@ -953,15 +972,18 @@ Http::Code AdminImpl::handlerQuitQuitQuit(absl::string_view, Http::HeaderMap&,
   return Http::Code::OK;
 }
 
-Http::Code AdminImpl::handlerListenerInfo(absl::string_view, Http::HeaderMap& response_headers,
+Http::Code AdminImpl::handlerListenerInfo(absl::string_view url, Http::HeaderMap& response_headers,
                                           Buffer::Instance& response, AdminStream&) {
-  response_headers.insertContentType().value().setReference(
-      Http::Headers::get().ContentTypeValues.Json);
-  std::list<std::string> listeners;
-  for (auto listener : server_.listenerManager().listeners()) {
-    listeners.push_back(listener.get().socket().localAddress()->asString());
+  const Http::Utility::QueryParams query_params = Http::Utility::parseQueryString(url);
+  const auto& it = query_params.find("format");
+
+  if (it != query_params.end() && it->second == "json") {
```

Comment:
Are all the handlers manually trying to figure out if `format=json`? I think we could factor this out to a separate function, this seems fragile and easily shared.

Reply:
Yeah, there are 3 instances of this in the file. I'll try factoring it out.

Reply:
Refactored the parsing logic out to a `formatParams` function. How does it look?